### PR TITLE
feat(mcp): add get_sales_order detail card

### PR DESCRIPTION
**Pull request by dougborg**
## Summary

Renders a read-only Prefab detail card for `get_sales_order`, matching how `get_item` / `get_variant_details` auto-render their cards. Previously `get_sales_order` returned plain JSON (`make_json_result`) with no card.

Follow-up to the storefront-link feature (#916); the shared `_render_ecommerce_link` helper built there is reused here.

Refs #913

## Card content (`build_so_detail_ui`)

Four-tier layout:

1. **Identity** — title as an external Link to the Katana SO page + status Badge (`status_badge_variant("sales_order", ...)`).
2. **Metrics** — order Total (`_format_money`), line-item count, delivery date as `Metric` tiles.
3. **Reference** — Customer + Location party lines (`_render_party_line`, resolved names), storefront deep-link (`_render_ecommerce_link`), scalar fields (customer ref / order date / picked date / notes / tracking), billing + shipping address blocks (`_render_address_block`), and a **static** line-item `DataTable` (SKU / Name / Qty / Unit Price / Line Total) with mustache `{{ so.rows }}` binding. Name-resolution cache-miss advisories surface as warning badges.
4. **Actions** — a single "View in Katana" link button.

## Response-model enrichment

`GetSalesOrderResponse` gains `customer_name` + `location_name`, resolved impl-side via `resolve_entity_name` against the typed cache (same pattern `create_sales_order` uses), plus a `warnings` list for cache-miss advisories. The card shows real names, never "Customer ID: 10". A null `location_id` short-circuits resolution (no lookup, no warning).

The tool fn now returns `make_tool_result(response, ui=build_so_detail_ui(...))` and registers with `meta=UI_META` so the card auto-renders on every call.

## Product decisions (per #913)

1. **Footer: "View in Katana" link only** — pure read card, no Fulfill/Edit/mutation buttons.
2. **Customer + location names resolved in this PR** via the typed cache.
3. **Line items: static inline DataTable**, no row-click drill-down — the SO is the destination.
4. **Auto-render** on every call (no preview/opt-in param), matching the other get_* tools.

## Test coverage

- `TestBuildSoDetailUI` (17 tests): happy path + edge cases — missing customer name (ID fallback), empty addresses, no storefront link, empty rows (friendly empty-state), warning badges, missing `katana_url` (no footer button), line-item cell formatting (null-SKU coalesce, display_name fallback, computed line total), footer = exactly one button.
- `get_sales_order` impl tests: customer/location name resolution, cache-miss advisories (2 warnings), null-location short-circuit (1 lookup), and the tool fn emitting a `$prefab` envelope in `structured_content`.
- Browser render scenario `so_detail` + render test (`test_so_detail_renders`) in `tests/browser/`.
- Synced the `help.py` resource for the detail card + resolved-name fields.

`uv run poe agent-check` clean (ruff/format/mdformat/ty); `uv run poe test` 3987 passed. The browser render test was validated in-process through the real render-server scenario dispatch (DataTable mounts with `{{ so.rows }}`, 2 state rows, single View-in-Katana button) — the headless Chromium run was blocked locally only because another worktree's `apps_dev` server held the shared ports; CI's `typescript-client`/browser job runs it cleanly.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: katana_mcp_server/src/katana_mcp/tools/prefab_ui.py [hunk 1]
```diff
@@ -7220,6 +7220,265 @@ def build_so_modify_ui(
...
+        unit = float_or_none(r.get("price_per_unit"))
+        total = float_or_none(r.get("total"))
+        if total is None and unit is not None and qty is not None:
+            total = unit * qty
+        cells.append(
+            {
+                "sku": r.get("sku") or "",
+                "display_name": (
+                    r.get("display_name") or f"Variant {r.get('variant_id')}"
+                ),
```
[the remark this answers is not in the document]
Comment (dougborg) [reply katana_mcp_server/src/katana_mcp/tools/prefab_ui.py]: Fixed — `_so_detail_row_cells` now computes the fallback as `f"Variant {variant_id}"` only when `variant_id` is not None, falling back to a neutral `"Unknown item"` otherwise, so a row with both `display_name` and `variant_id` null no longer renders `"Variant None"`. Added a regression test (`test_null_variant_id_row_uses_neutral_label`). Squashed into the feat commit.
